graphql_compiler/post_processing/sql_post_processing.py

```diff
@@ -0,0 +1,128 @@
+        result.replace("^d", "|") if result is not None else None for result in list_result
+    ]
+
+    # Convert "^n" to "~".
+    list_result = [
+        result.replace("^n", "~") if result is not None else None for result in list_result
+    ]
+
+    # Convert "^e" to "^".
+    list_result = [
+        result.replace("^e", "^") if result is not None else None for result in list_result
+    ]
+
+    # Convert "&#x{2 digit HEX};" to unicode character.
+    new_list_result: List[Optional[str]] = []
+    for result in list_result:
+        if result is not None:
+            split_result = re.split("&#x([A-Fa-f0-9][A-Fa-f0-9]);", result)
+            new_result = split_result[0]
+            for hex_value, next_substring in zip(split_result[1::2], split_result[2::2]):
+                new_result += chr(int(hex_value, 16)) + next_substring
+            new_list_result.append(new_result)
+        else:
+            new_list_result.append(None)
+
+    # Convert "&amp;" to "&", "&gt;" to ">", "&lt;" to "<".
+    list_result = [
+        html.unescape(result) if result is not None else None for result in new_list_result
+    ]
+
+    # Convert to the appropriate return type.
+    list_result_to_return: List[Optional[Any]] = [
+        list_entry_type.parse_value(result) if result is not None else None
+        for result in list_result
+    ]
+
+    return list_result_to_return
+
+
+def post_process_mssql_folds(
+    schema_info: SQLAlchemySchemaInfo, graphql_query: str, query_results: List[Dict[str, Any]]
+) -> None:
+    r"""Convert XML PATH fold results from a string to a list of the appropriate type.
+
+    See _get_xml_path_clause in graphql_compiler/compiler/emit_sql.py for an in-depth description
+    of the encoding process.
+
+    Post-processing steps:
+        1. split on "|",
+        2. convert "~" to None
+        3. convert caret escaped characters (excluding "^" itself)
+            i.e. "^d" (delimiter) to "|" and "^n" (null) to "~"
+        4. with caret escaped characters removed, convert "^e" to "^"
+        5. convert ampersand escaped characters (excluding "&" itself)
+            i.e. "&gt;" to ">", "&lt;" to "<" and "&#xHEX;" to "\xHEX"
+        6. with ampersand escaped characters removed, convert "&amp;" to "&"
+
+    Args:
+        schema_info: SQLAlchemySchemaInfo, schema the query was run on
+        graphql_query: str, a GraphQL query
+        query_results: Dict[str, Any], results from graphql_query being run with schema_info,
+                       mutated in place
+
+    """
+    # Get the internal representation of the query.
+    ir_and_metadata = graphql_to_ir(
+        schema_info.schema,
+        graphql_query,
+        type_equivalence_hints=schema_info.type_equivalence_hints,
+    )
+    lowered_ir_blocks = sql_backend.lower_func(schema_info, ir_and_metadata)
+
+    # Loop through ir_blocks; for all ConstructResult for FoldedContextFields, look up string result
+    # in query_results and update to a list.
```

Comment:
The output metadata is intuitively the sensible thing to use because:
- It's made for this purpose. If it's not serving it's purpose, we should improve it instead of avoiding it if it's not too much work.
- It's easier to understand. The IR is hard, and you don't want to assume IR knowledge to write post-processing steps.

The problem with the output metadata for this purpose is that it only knows which outputs are list type, and can't differentiate outputting list typed properties from outputting folded non-list properties. I think it's worth amending the output metadata with this information.